protecode/scanning.py

```diff
@@ -137,20 +138,24 @@ def iter_vulnerabilities_with_assessments(
                     continue
                 yield component, vulnerability, tuple(vulnerability.triages())
 
-        # only consider the most recent 50 products to import assessments from
-        # assumption: assessments are copied anyways from older assessments to
-        # newer ones so considering old assessments won't have any extra value
-        products_to_import_from.sort(
-            key=lambda p: p.product_id(),
-            reverse=True,
-        )
-        products_to_import_from = products_to_import_from[:50]
-
-        # do a shuffle to increase cache hits as otherwise same product ids are
-        # more likely to be handled simultaneously among the different threads
-        random.shuffle(products_to_import_from)
-
         for product in products_to_import_from:
+            if delete_inactive_products_after:
+                if deletion_after := product.custom_data().get('DELETION_AFTER'):
+                    if datetime.datetime.now() > datetime.datetime.fromisoformat(deletion_after):
```

Comment:
preferably use dateutil.parser - datetime.datetime's fromisoformat is (I think) buggy and does not properly parse all valid iso8601-timestamps (dateutil.parser is more robust and a drop-in replacement, so I recommend we use it instead of datetime.datetime.isoparse always)